### PR TITLE
 maps: keep only one active map with given name

### PR DESCRIPTION
**Pull request by rmycek**

### CODE REVIEW

File: src/main/java/com/guysfromusa/carsgame/entities/MapEntity.java [hunk 1]
```diff
@@ -31,10 +33,14 @@
...
-    private boolean deleted;
+    private String active = ACTIVE;
 
     public MapEntity(String name, String content) {
         this.name = name;
         this.content = content;
     }
+
+    public void deactivate() {
+        active = Long.toString(id);
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/entities/MapEntity.java]: why id?

File: src/main/java/com/guysfromusa/carsgame/entities/MapEntity.java [hunk 2]
```diff
@@ -8,11 +8,13 @@
 import javax.persistence.*;
 
 @Entity
-@Table(name = "MAP")
+@Table(name = "MAP", uniqueConstraints = {@UniqueConstraint(columnNames = {"name", "active"})})
 @NoArgsConstructor
 @ToString
 public class MapEntity {
 
+    public static final String ACTIVE = "ACTIVE";
```
Comment (sebastian-mikucki) [inline src/main/java/com/guysfromusa/carsgame/entities/MapEntity.java]: can this be private ?

File: src/main/java/com/guysfromusa/carsgame/entities/MapEntity.java (hunk 1 above)
Comment (rmycek) [reply src/main/java/com/guysfromusa/carsgame/entities/MapEntity.java]: constraint is on 2 columns (name, active)
active column has magic value "ACTIVE"
if map is mark as deleted active is set to not "ACTIVE" value, could be anything
but id I am sure not to fail on constraint

File: src/main/java/com/guysfromusa/carsgame/entities/MapEntity.java (hunk 2 above)
Comment (rmycek) [reply src/main/java/com/guysfromusa/carsgame/entities/MapEntity.java]: no, used in code

**Pull request merged**